magento/magento2
# Removed workaround for old Webkit bug in the TinyMCE editor for selec…

…tions of images and hr elements, because this workaround now causes problems in Google Chrome version 58.

### Description
This PR attempts to solve a new bug which popped up by upgrading Google Chrome to version 58.
In the TinyMCE editor, we can no longer select images (or widgets, which are represented by an image), so we can no longer edit images (or widgets) using the TinyMCE editor.
See https://github.com/magento/magento2/issues/9518 for more detailed info with links to a bunch of external sources.

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/9518: Chrome version 58 causes problems with selections in the TinyMCE editor

### Manual testing scenarios
1. Start editing a CMS page in the backend of Magento using Google Chrome version 58
2. Make sure you have a CMS page which contains an image and/or a widget
3. Try to select the image or the widget, by clicking it
4. You won't succeed, because a JS error is being thrown: `Failed to execute 'setBaseAndExtent' on 'Selection': There is no child at offset 1.`

### Contribution checklist

### Discussion
I only briefly tested these changes, only on Chrome 58 and Safari 10.1, so it would be highly appreciated if someone else can test all browsers which Magento supports and see if this doesn't break any other browser.

I applied the changes to all tiny_mce scripts I could find, although I think only the `tiny_mce_src.js` one is used in the backend of Magento. But since you bundle all the other scripts in Magento, It could be possible that someone is using another script, so I assumed this should be fixed in all the tiny_mce scripts.
Also: the minified tiny_mce scripts were manually edited, I don't know the command which was being used to generate the minified scripts, so please double check that my manual edits are ok.

Since this bug will probably popup with a lot of shopowners in the near future, this fix should get a bit of a higher priority and should also get backported to Magento versions 2.1 and 2.0 as soon as possible.
If this PR is accepted, I can create a backport for version 2.1, but not sure how backporting to 2.0 is supposed to work?
